### ISSUE TITLE
A Node Note

### ISSUE THREAD

**Issue by okjim3ML** [atuttle/mxunit-watch#10]
I was getting mxunit to initialize properly but then it would fail the first time I asked it to respond to a change in the watched directory. I would get a node error on the call to test-list.cfm, even though the same call was successful seconds ago. The node error was this:

{ [Error: connect EMFILE] code: 'EMFILE', errno: 'EMFILE', syscall: 'connect' }

There's lots of info about this on stack overflow and it seems to be related to all the files in the watch directory. I learned by running "ulimit -a" that my limit was 256. I then ran "ulimit -n 10000" to bump my limit up to 10k. This fixed the problem, but only for this bash session. You can go here to learn to make the adjustment permanent, at least on mac and linux:

[http://stackoverflow.com/questions/34588/how-do-i-change-the-number-of-open-files-limit-in-linux](http://stackoverflow.com/questions/34588/how-do-i-change-the-number-of-open-files-limit-in-linux)

I've run other watch programs before such as grunt-contrib-watch and not run into this, so I wonder what they do differently. Ultimately it was an OS config setting that caused the issue, but I thought I would post it here because it got in the way of my using mxunit-watch. I'm enjoying using it now, so thanks for creating it.

**Comment by seancoyne** [comment]
What is your configuration for mxunit-watch?  You can configure what it watches so you don't need to watch all your files.  For example, for most of my projects, I only watch CFCs, but mxunit-watch will watch CFM, CFC and XML by default, if I remember correctly.

There is also a grunt plugin for mxunit-watch so you can use grunt-contrib-watch to do the actual watching if you prefer.

**Comment by okjim3ML** [comment]
Yes, I did reduce the scope of the watch at the same time as I bumped up the ulimit. I didn't mention that b/c bumping the files to 10k fixed the problem with overkill. I was originally watching the root directory of my application so that I would catch any changes to application.cfc. I'm quite sure I don't have more than 256 files that are either .cfm, .cfc, or xml. This project is primarily front-end javascript with  coldfusion providing a data api. I think the watcher is watching ALL the files in the directory and taking action only when a matching file is changed. That's the only way I could have gotten over 256. All those js files in node_modules directory. I do plan to roll it into grunt-contrib-watch, but I thought it would be simpler to get it running on its own first. On the other hand it might be nicer to have the cf tests separate from the js tests. Some days I don't do anything on the cf side and some days I don't do anything on the js side.
